### PR TITLE
Keras Optimizer Documentation Fix

### PR DESCRIPTION
**Pull request by DEKHTIARJonathan**
The documentation of Keras Optimizer mention the following:
```python
If global_step  was not `None`, that operation also increments global_step.
```

You can also find it here: https://www.tensorflow.org/versions/r2.0/api_docs/python/tf/keras/optimizers/Optimizer#apply_gradients

Keras Optimizer do not use the concept of global_step. It's misleading and shall be removed.

**Note:** As it impacts directly the TF2.0 documentation. It would be nice if this PR could be cherrypicked in the next beta/official release for TF2.0

### CODE REVIEW

File: tensorflow/python/keras/optimizer_v2/optimizer_v2.py [hunk 1]
```diff
@@ -304,8 +304,7 @@ def minimize(self, loss, var_list, grad_loss=None, name=None):
       name: Optional name for the returned operation.
 
     Returns:
-      An Operation that updates the variables in `var_list`.  If `global_step`
-      was not `None`, that operation also increments `global_step`.
+      An Operation that updates the variables in `var_list`.
```
Comment (tanzhenyu) [inline tensorflow/python/keras/optimizer_v2/optimizer_v2.py]: that applies gradients and increase local step.

**Comment by DEKHTIARJonathan** [comment]
Local step not global step. Especially that global step refers to the concept of TF 1.x which doesn't exist anymore in TF 2.x

**Comment by tanzhenyu** [comment]
Is that a response, or a comment?

**Comment by DEKHTIARJonathan** [comment]
A response to your comment 😊

**Comment by tanzhenyu** [comment]
Hmmm...my comment is to add "increase local step xxx"....

**Comment by DEKHTIARJonathan** [comment]
Oh I misunderstood your point then.
I think the optimizer local step needs to be explained better than this. Giving a reference to an attribute not referenced at any other point around is nothing but confusing.
Nonetheless I agree would be nice to mention that it increases an internal counter.
Maybe you wish to suggest a direction? Want me to add you as a contributor to my fork?

**Comment by tanzhenyu** [comment]
> Oh I misunderstood your point then.
> I think the optimizer local step needs to be explained better than this. Giving a reference to an attribute not referenced at any other point around is nothing but confusing.
> Nonetheless I agree would be nice to mention that it increases an internal counter.
> Maybe you wish to suggest a direction? Want me to add you as a contributor to my fork?

I think it's fine to mention it in this PR (where I suggested). Optimizer has both setter and getter for iteration so it's not a hidden concept.
Though you could name it to be `iterations` instead of `local step`.

**Comment by mellanox-github** [comment]
Can one of the admins verify this patch?

**Comment by tanzhenyu** [comment]
There is another request to fix this. So I made a code change for it instead of waiting here. @DEKHTIARJonathan

**Comment by DEKHTIARJonathan** [comment]
I suppose you talk about commit 97fb325e3b8499d375251359fd69abd2fa96ee39.
Good with me 👍
